recipes/default.rb

```diff
@@ -14,10 +14,18 @@
 # limitations under the License.
 #
 
+ruby_block 'ensure node can resolve API FQDN' do
+  extend ChefAnalyticsCookbook::Helpers
+  block { repair_api_fqdn }
+  only_if { api_fqdn_available? }
+  not_if { api_fqdn_resolves? }
+end
```

Comment:
Okay I was reading it wrong yesterday, its not mutating the attribute at all its just creating a file_edit resource out of a ruby_block and Chef::Utils::FileEdit.  That is also bad, but not on the scale of what I thought I was reading yesterday, and I don't have any better fixes.